pkg/ottl/expression.go

```diff
@@ -549,6 +550,89 @@ func (g StandardIntLikeGetter[K]) Get(ctx context.Context, tCtx K) (*int64, erro
 	return &result, nil
 }
 
+// ByteSliceLikeGetter is a Getter that returns []byte by converting the underlying value to an []byte if necessary
+type ByteSliceLikeGetter[K any] interface {
+	// Get retrieves []byte value.
+	// The expectation is that the underlying value is converted to []byte if possible.
+	// If the value cannot be converted to []byte, nil and an error are returned.
+	// If the value is nil, nil is returned without an error.
+	Get(ctx context.Context, tCtx K) (byteSlice, error)
+}
+
+type StandardByteSliceLikeGetter[K any] struct {
+	Getter func(ctx context.Context, tCtx K) (any, error)
+}
+
+//nolint:revive
+func (g StandardByteSliceLikeGetter[K]) Get(ctx context.Context, tCtx K) (byteSlice, error) {
+	val, err := g.Getter(ctx, tCtx)
+	if err != nil {
+		return nil, fmt.Errorf("error getting value in %T: %w", g, err)
+	}
+	if val == nil {
+		return nil, nil
+	}
+	var result byteSlice
+	switch v := val.(type) {
+	case byteSlice:
+		result = v
+	case []byte:
+		result = v
+	case string:
+		result = []byte(v)
+	case float64:
+		result = float64ToBytes(v)
+	case int64:
+		result = int64ToBytes(v)
+	case bool:
+		if v {
+			result = []byte("1")
+		} else {
+			result = []byte("0")
+		}
+	case pcommon.Value:
+		switch v.Type() {
+		case pcommon.ValueTypeBytes:
+			result = v.Bytes().AsRaw()
+		case pcommon.ValueTypeInt:
+			result = int64ToBytes(v.Int())
+		case pcommon.ValueTypeDouble:
+			result = float64ToBytes(v.Double())
+		case pcommon.ValueTypeStr:
+			result = []byte(v.Str())
+		case pcommon.ValueTypeBool:
+			if v.Bool() {
+				result = []byte("1")
+			} else {
+				result = []byte("0")
+			}
```

Comment:
```suggestion
			if v.Bool() {
				result = []byte{1}
			} else {
				result = []byte{0}
			}
```
I think as-written these will return the UTF-8 sequences for each character.